pkg/cvo/status.go

```diff
@@ -379,6 +380,21 @@ func updateClusterVersionStatus(cvStatus *configv1.ClusterVersionStatus, status
 		}
 	}
 
+	if enabledGates.ResourceReconciliationIssuesCondition {
+		rriCondition := configv1.ClusterOperatorStatusCondition{
+			Type:    resourceReconciliationIssuesConditionType,
+			Status:  configv1.ConditionFalse,
+			Reason:  noResourceReconciliationIssuesReason,
+			Message: noResourceReconciliationIssuesMessage,
+		}
+		if status.Failure != nil {
+			rriCondition.Status = configv1.ConditionTrue
+			rriCondition.Reason = resourceReconciliationIssuesFoundReason
+			rriCondition.Message = fmt.Sprintf("%s: %s", resourceReconciliationIssuesFoundMessage, status.Failure.Error())
+		}
+		resourcemerge.SetOperatorStatusCondition(&cvStatus.Conditions, rriCondition)
+	}
```

Comment:
else remove `resourceReconciliationIssuesConditionType`?  So we don't leak stale data if the gate is disabled?  But also bumps up against [the possibility of races during the update causing brief periods of instability in gate enablement][1], and I could see logic that tried to smooth that kind of thing out while we were de-racing.

[1]: https://github.com/openshift/cluster-version-operator/pull/1032#discussion_r1476406296

Reply:
Addressed in ef62981b893e179049d367a05d8f6cb2f7be3dd5: removing when the feature gate is known to be disabled, maintaining when it is known to be enabled, or when the flags are unknown but we find the existing condition in the cluster